src/motion/cg_optimizer.F

```diff
@@ -85,12 +101,51 @@ RECURSIVE SUBROUTINE geoopt_cg(force_env, gopt_param, globenv, geo_section, &
       output_unit = cp_print_key_unit_nr(logger, geo_section, "PRINT%PROGRAM_RUN_INFO", &
                                          extension=".geoLog")
       CALL print_geo_opt_header(gopt_env, output_unit, "CONJUGATE GRADIENTS")
-      CALL cp_cg_main(force_env, x0, gopt_param, output_unit, globenv, &
-                      gopt_env, do_update=my_do_update)
+
+      ! [pcazade] find space_group
+      CALL force_env_get(force_env, subsys=subsys)
+      keep_sg = .FALSE.
+      CALL section_vals_val_get(geo_section, "KEEP_SPACE_GROUP", l_val=keep_sg)
+      IF (keep_sg) THEN
+         SELECT CASE (gopt_env%type_id)
+         CASE (default_minimization_method_id, default_ts_method_id)
+            CALL force_env_get(force_env, subsys=subsys)
+            CALL identify_space_group(spgr, subsys, subsys%cell, geo_section, gopt_env, output_unit)
+            CALL spgr_apply_rotations_coord(spgr, x0)
+            CALL print_spgr(spgr)
+         CASE (default_cell_method_id)
+            SELECT CASE (gopt_env%cell_method_id)
+            CASE (default_cell_direct_id)
+               CALL force_env_get(force_env, subsys=subsys)
+               CALL identify_space_group(spgr, subsys, subsys%cell, geo_section, gopt_env, output_unit)
+               CALL spgr_apply_rotations_coord(spgr, x0)
+               CALL print_spgr(spgr)
+            CASE (default_cell_geo_opt_id, default_cell_md_id)
+               keep_sg = .FALSE.
+            CASE DEFAULT
+               keep_sg = .FALSE.
+            END SELECT
+         CASE DEFAULT
+            keep_sg = .FALSE.
```

Comment:
I think, here you should call `CPABORT("KEEP_SPACE_GROUP not implemented for method foo")` instead.

Reply:
I am not sure CPABORT is what we need here. Correct me if I am wrong but, don't one end up in the CASE(default_cell_geo_opt_id) during the cell optimization step when one is not using the direct cell method but rather alternate cell opt and geo opt. If it is the case, the routines for space group symmetry have to be disabled for this particular step, and enabled for the geo_opt one.

Reply:
However, for default_cell_md_id, it makes sense to abort, though.

Reply:
You are right, that would probably break the legacy cell opt. However, I doubt anybody is still using it and we're planing to remove it (#1233).

Unfortunately, our motion code is pretty bad shape. I guess, you could tell from the fact that you had to hook into each optimizer (CG, BFGS, LBFGS) separately.

So, I'd rather error on the side of caution. Are you sure it's ok to continue with `keep_sg = .FALSE.` in all those cases?

Reply:
As far as I can tell and test, it is. Furthermore, space_group_symmetry works for the legacy cell opt. I made sure of it. I see not harm keeping this until the feature is removed. I will split the CASE however to have a CPABORT for the md case.

Reply:
Sounds good! You certainly know the code better than me at this point.